blockprod/src/detail/mod.rs

```diff
@@ -73,6 +83,57 @@ impl BlockProductionInterface for BlockProduction {
     fn is_connected(&self) -> bool {
         !self.builder_tx.is_closed()
     }
+
+    async fn generate_block(
+        &self,
+        reward_destination: Destination,
+        signed_transactions: Vec<SignedTransaction>,
+    ) -> Result<Id<Block>, BlockProductionError> {
+        let (best_block_id, best_block_height, accumulator) = self
+            .chainstate_handle
+            .call(|this| {
+                // Stay with DefaultTxaccumulator as we are being
+                // explicit with the transactions to build blocks with
+                let mut accumulator =
+                    DefaultTxAccumulator::new(this.get_chain_config().max_block_size_from_txs());
+
+                for signed_transaction in signed_transactions {
+                    if accumulator.add_tx(signed_transaction, Fee::new(Amount::ZERO)).is_err() {
+                        return None;
+                    };
+                }
+
+                if let Ok(block_id) = this.get_best_block_id() {
+                    if let Ok(Some(block_height)) = this.get_block_height_in_main_chain(&block_id) {
+                        return Some((block_id, block_height, accumulator));
+                    }
+                }
+
+                None
+            })
+            .await?
+            .ok_or(BlockProductionError::SubsystemCallError(
+                CallError::SubmissionFailed,
+            ))?;
+
+        let (_tx, dummy_rx) = crossbeam_channel::unbounded();
+
+        let block_maker = BlockMaker::new(
+            self.chain_config.clone(),
+            self.chainstate_handle.clone(),
+            self.mempool_handle.clone(),
+            self.time_getter.clone(),
+            reward_destination,
+            best_block_id,
+            best_block_height,
+            dummy_rx,
+        );
+
+        // TODO: Stop any current block makers before making this new
```

Comment:
I wouldn't put it like this, because the operation of a block maker is more complex than this. Usually, there shouldn't be block makers running if you call generate.